src/System.Text.Json/src/System/Text/Json/Utf8JsonReader.MultiSegment.cs

```diff
@@ -0,0 +1,2408 @@
+            // Those cases are captured in ConsumeNextToken and ConsumeValue.
+            if (first <= JsonConstants.Space)
+            {
+                SkipWhiteSpaceMultiSegment();
+                if (!HasMoreDataMultiSegment())
+                {
+                    goto Done;
+                }
+                first = _buffer[_consumed];
+            }
+
+            if (_tokenType == JsonTokenType.None)
+            {
+                goto ReadFirstToken;
+            }
+
+            if (first == JsonConstants.Slash)
+            {
+                retVal = ConsumeNextTokenOrRollbackMultiSegment(first);
+                goto Done;
+            }
+
+            if (_tokenType == JsonTokenType.StartObject)
+            {
+                if (first == JsonConstants.CloseBrace)
+                {
+                    EndObject();
+                }
+                else
+                {
+                    if (first != JsonConstants.Quote)
+                    {
+                        ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedStartOfPropertyNotFound, first);
+                    }
+
+                    long prevTotalConsumed = _totalConsumed;
+                    int prevConsumed = _consumed;
+                    long prevPosition = _bytePositionInLine;
+                    long prevLineNumber = _lineNumber;
+                    retVal = ConsumePropertyNameMultiSegment();
+                    if (!retVal)
+                    {
+                        // roll back potential changes
+                        _consumed = prevConsumed;
+                        _tokenType = JsonTokenType.StartObject;
+                        _bytePositionInLine = prevPosition;
+                        _lineNumber = prevLineNumber;
+                        _totalConsumed = prevTotalConsumed;
+                    }
+                    goto Done;
+                }
+            }
+            else if (_tokenType == JsonTokenType.StartArray)
+            {
+                if (first == JsonConstants.CloseBracket)
+                {
+                    EndArray();
+                }
+                else
+                {
+                    retVal = ConsumeValueMultiSegment(first);
+                    goto Done;
+                }
+            }
+            else if (_tokenType == JsonTokenType.PropertyName)
+            {
+                retVal = ConsumeValueMultiSegment(first);
+                goto Done;
+            }
+            else
+            {
+                retVal = ConsumeNextTokenOrRollbackMultiSegment(first);
+                goto Done;
+            }
+
+            retVal = true;
+
+        Done:
+            return retVal;
+
+        ReadFirstToken:
+            retVal = ReadFirstTokenMultiSegment(first);
+            goto Done;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Really?

Reply:
Will remove. This method grew larger than what I initially had it as.